File: 兴趣班入门百练讲义/NQ100/NQ023/NQ023_improved.cpp
// NQ023 桌球比赛
#include <iostream>
using namespace std;
int main() {
    int m;
    while(cin>>m){
        int r=0,y=0;
        char c;
        while(m--){
            cin>>c;
            if(c=='R')r++;  // 统计红球
            else if(c=='Y')y++;  // 统计黄球
            else if(c=='B')cout<<(r==7?"Red":"Yellow")<<'\n';  // B指令
            else if(c=='L')cout<<(y==7?"Yellow":"Red")<<'\n';  // L指令
        }
    }
    return 0;
}

File: source/algorithm/NQ023/NQ023_improved.py
# NQ023 桌球比赛
import sys
def main():
    s=[]
    for l in sys.stdin:
        l=l.strip()
        if l:s.append(l) if l.isdigit() else s.extend(list(l))
    p=0
    while p<len(s):
        if not s[p].isdigit():
            p+=1
            continue
        m=int(s[p]);p+=1
        if m<=0:continue
        r,y=0,0
        for _ in range(m):
            if p>=len(s):break
            c=s[p];p+=1
            if c=='R':r+=1
            elif c=='Y':y+=1
            elif c=='B':print("Red" if r==7 else "Yellow")
            elif c=='L':print("Yellow" if y==7 else "Red")
if __name__=="__main__":main()